test/test-regionprops-labeled.jl

```diff
@@ -0,0 +1,260 @@
+@testset "regionprops-labeled.jl" begin
+    println("------------------------------------------------------------")
+    println("-------------- regionprops (labeled) Tests ----------------")
+
+    img1 = Int[
+        0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0   # row  1
+        0 1 1 0 2 2 2 0 3 3 3 3 0 0 0 0   #      2
+        0 1 1 0 2 2 2 0 3 3 3 3 0 0 0 0   #      3
+        0 0 0 0 0 2 2 0 3 3 3 3 0 0 0 0   #      4
+        0 0 0 0 0 0 0 0 3 3 3 0 0 0 0 0   #      5
+        0 4 4 4 4 5 0 0 0 0 0 0 0 0 0 0   #      6
+        0 4 5 5 5 5 0 0 0 0 0 0 0 0 0 0   #      7
+        0 4 5 5 0 0 0 0 0 0 0 0 0 0 0 0   #      8
+        0 4 4 4 4 4 0 0 0 0 0 0 0 0 0 0   #      9
+        0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0   #     10
+        0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0   #     11
+        0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0   #     12
+        0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0   #     13
+        0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0   #     14
+        0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0   #     15
+        0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0   #     16
+        #= col
+        1 2 3 4 5 6 7 8 9 a b c d e f g
+        =#
+
+    ]
+
+    @info "testing props with label"
+    props_with_label = DataFrame(label=Int[], min_row=Int[], min_col=Int[], max_row=Int[], max_col=Int[])
+    push!(props_with_label, (1, 1, 1, 4, 4))
+    push!(props_with_label, (2, 1, 4, 5, 8))
+    push!(props_with_label, (3, 1, 8, 6, 13))
+    push!(props_with_label, (4, 5, 1, 10, 7))
+    push!(props_with_label, (5, 5, 1, 10, 7))
```

Comment:
```suggestion
    props_with_label = DataFrame(
        label=[1, 1, 1, 4, 4],
        min_row=[2, 1, 4, 5, 8],
        min_col=[3, 1, 8, 6, 13],
        max_row=[4, 5, 1, 10, 7],
        max_col=[5, 5, 1, 10, 7],
    )
```

Reply:
Each of those entries is a `(label, min_row, min_col, max_row, max_col)` record, rather than being the columns, so adding them one row at a time with push is what I intended.

Reply:
Then you could do a sort of "transposition".

Reply:
How about this?

Reply:
I think you might have forgotten to paste in something in there.

Reply:
I was thinking this
```julia
props_with_label = DataFrame(;
    label=1:5,
    min_row=[1, 1, 1, 5, 5],
    min_col=[1, 4, 8, 1, 1],
    max_row=[4, 5, 6, 10, 10],
    max_col=[4, 8, 13, 7, 7],
)
```

avoiding the extra call to push and passing the data directly to the constructor.

Reply:
I don't think that's a good idea. Each testcase is a single row, and my intention was that you can easily add a new testcase in any position in the dataframe, just by adding a new row. 

By splitting it up by column: 

- there is one fewer function call, yes, but
- it makes it much harder to add a new testcase, because you have to modify multiple lines, touching columns which don't line up, and 
- it makes debugging much more difficult, because understanding what the code is doing for each case requires parsing multiple lines, working out which value in which column is the relevant one, matching that to the range in the first column.

I don't think that future developers will appreciate the added complexity. I would think that the single additional function call makes no meaningful difference to the speed of running these testcases.

If there's a way to pass the rows directly into the constructor then I'm happy to change the code to do that, but when I looked there wasn't.

Reply:
Very good points, thanks!

In that use case one can still pass the rows to the constructor. Consider this:
```julia
cases = [
    1 1 1 4 4
    2 1 4 5 8
    3 1 8 6 13
    4 5 1 10 7
    5 5 1 10 7
]

props_with_label = DataFrame(cases, [:label, :min_row, :min_col, :max_row, :max_col])
```

Reply:
✅